docs/dc/simple_construction/index.rst

```diff
@@ -0,0 +1,51 @@
+.. skip: start
+
+===================
+Simple Construction
+===================
+
+Before looking at injection, let's take a look at how dataclasses get registered. We'll use a very simple example: a ``Greeter`` that greets people, for example in a grocery store.
+
+First, imagine we have a nice, boring ``Greeter``:
+
+.. literalinclude:: models.py
+
+Since this is a ``wired`` application, we need an application with a registry. Dataclasses are fun, let's use them for our application:
+
+.. literalinclude:: app.py
+
+During startup, we create the app (and thus registry) and then start configuring our stuff. For example, here is a configuration step:
+
+.. literalinclude:: configure.py
+
+If we were using ``wired`` directly, we would have to define a factory function and register it:
+
+.. code-block:: python
+
+    # NO LONGER NEED THIS, wired.dataclasses makes a factory
+    def greeter_factory(container):
+        greeter = Greeter()
+        return greeter
+
+    registry.register_factory(greeter_factory, Greeter)
+
+But with ``wired.dataclasses``, no factory function is needed, as `wired.dataclasses` makes a factory for you:
+
+.. literalinclude:: configure.py
```

Comment:
It looks like `configure.py` is included twice in this file and I don't think the contents is right as 1) it's not using/importing the `App` object and 2) it's making an unused container which probably shouldn't be there. Can you clarify for me what's going on here?

Reply:
You're right, I removed the duplicate. The reason it doesn't seem to be used is a trick I did for doc purposes: the block at the bottom that is included in the docs, actually comes from the test.

But I didn't want the docs to show mega-long imports and I couldn't get relative working right.

I made a small adjustment for this, to show `configure.register`.